### PR TITLE
CI Workflow: Update sphinx-intl new release & Install Transifex CLI.

### LINKED ISSUE #5748: Update the CI workflow for transifex upload when new sphinx-intl ready
As mentioned in https://github.com/aiidateam/aiida-core/pull/5741#discussion_r1013387517, the sphinx-intl adopt the same API to transifex and use the binary compiled from Golang which the `tx` executable is not installed as extra requirement anymore. 
But by:

```
      - name: Install Transifex CLI
        run: |
          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
          mv tx /usr/local/bin/tx
```

We need to adopt it after the new sphinx-intl release comes out.

### PR DESCRIPTION
**Pull request by AhmedBasem20**
CLoses #5748 

The new sphinx-intl release (2.1.0) is now updated, Also Transfix CLI installation script has been added.

### CODE REVIEW

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks for the contribution @AhmedBasem20,  just one minor request. Can you also add a version for the transifex cli as mentioned in https://github.com/transifex/cli#installing-with-a-script-linuxmac? 

I think there is some problem on my initial implementation, since I now realize it is not a good practice to push to transifex when new version released. The documentation `.pot` files need to updated when there are new changes in `/doc` folder in the main branch, which will make the `latest` of other language versions of documentation synchronize with the latest official version. 
But nothing related to this PR, and I'll leave it to myself in the future.

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -39,6 +39,11 @@ jobs:
       run:
         make -C docs gettext
 
+    - name: Install Transifex CLI
+      run: |
+        curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash
```
Comment (unkcpz) [inline .github/workflows/post-release.yml:44]:
```suggestion
        curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash -s -- v1.6.5
```

**Comment by AhmedBasem20** [comment]
Thanks @unkcpz for reviewing. I added the requested transifex cli version. Your insight on best practices is greatly appreciated. Thanks again for your time and guidance!

**Review by unkcpz — APPROVED** [review]
Thanks @AhmedBasem20. I give it a local test, all good!

**Pull request merged**